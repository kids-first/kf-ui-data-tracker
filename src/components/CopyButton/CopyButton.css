.CopyButton {
  @extend .inline-block, .relative, .text-xs;
}

.CopyTooltip {
  top: 150%;
  left: 50%;
  border-radius: 4px;
  @extend .absolute, .w-16, .z-10, .-ml-8, .bg-black, .text-white, .text-xs,
    .text-center, .rounded-sm, .py-8;
}

.CopyTooltip:after {
  content: '';
  bottom: 100%;
  left: 50%;
  margin-left: -5px;
  border-width: 5px;
  border-color: transparent transparent black transparent;
  @extend .absolute, .border-solid;
}
