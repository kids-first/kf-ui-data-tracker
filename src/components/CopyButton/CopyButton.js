import React, {useState} from 'react';
import PropTypes from 'prop-types';
import {Button, Icon, Popup} from 'semantic-ui-react';
import {CopyToClipboard} from 'react-copy-to-clipboard';

/**
 * A simple button that is wrapped with an icon and a popup prompting the user
 * to copy the provided text.
 * Passes additional props to the Button
 */
const CopyButton = ({text, ...props}) => {
  const [copied, setCopied] = useState(false);
  return (
    <Popup
      inverted
      position="top left"
      size="mini"
      className='CopyButton--popup'
      trigger={
        <CopyToClipboard
          text={text}
          onCopy={() => {
            setCopied(true);
            setTimeout(() => {
              setCopied(false);
            }, 700);
          }}
        >
          <Button
            {...props}
            icon
            labelPosition="left"
            onClick={e => e.stopPropagation()}
          >
            <Icon name="copy" />
            {text}
          </Button>
        </CopyToClipboard>
      }
    >
      {copied ? <Icon color="green" name="check" /> : 'Copy'}
    </Popup>
  );
};

CopyButton.propTypes = {
  /** The string to be copied */
  text: PropTypes.string.isRequired,
};

CopyButton.defaultProps = {
  text: null,
};

export default CopyButton;
