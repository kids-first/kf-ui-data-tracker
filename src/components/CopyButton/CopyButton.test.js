import React from 'react';
import {act} from 'react-dom/test-utils';
import {render, fireEvent} from 'react-testing-library';
import CopyButton from './CopyButton';

/* Sets up some browser functions that are needed for react-copy-to-clipboard
 * that do not exist in the test renderer
 */
beforeAll(() => {
  // Always confirm prompts
  global.prompt = () => true;
  // Print alerts to console
  window.alert = () => true;

  // Needed to allow react-copy-to-clipboard to work in test renderer
  const getSelection = () => ({
    rangeCount: 0,
    addRange: () => {},
    getRangeAt: () => {},
    removeAllRanges: () => {},
  });

  window.getSelection = getSelection;
  document.getSelection = getSelection;
});

it('renders correctly', () => {
  const tree = render(<CopyButton text="testing" />);
  expect(tree).toMatchSnapshot();
});
