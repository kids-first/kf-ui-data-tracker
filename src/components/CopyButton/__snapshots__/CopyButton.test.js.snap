// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays checkmark svg on click 1`] = `
<span
  class=""
>
  <button>
    <svg
      height="10"
      id="mock-icon"
      kind="copy"
      width="10"
    >
      <title>
        icon-
        copy
      </title>
    </svg>
  </button>
</span>
`;

exports[`displays checkmark svg on click 2`] = `
<span
  class=""
>
  <button>
    <span
      class="CopyButton"
    >
      <svg>
        <title>
          icon-checkmark
        </title>
      </svg>
      <span
        class="CopyTooltip"
      >
        Copied!
      </span>
    </span>
  </button>
</span>
`;

exports[`displays checkmark svg on click 3`] = `
<span
  class=""
>
  <button>
    <svg
      height="10"
      id="mock-icon"
      kind="copy"
      width="10"
    >
      <title>
        icon-
        copy
      </title>
    </svg>
  </button>
</span>
`;

exports[`renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        class=""
      >
        <button>
          <svg
            height="10"
            id="mock-icon"
            kind="copy"
            width="10"
          >
            <title>
              icon-
              copy
            </title>
          </svg>
        </button>
      </span>
    </div>
  </body>,
  "container": <div>
    <span
      class=""
    >
      <button>
        <svg
          height="10"
          id="mock-icon"
          kind="copy"
          width="10"
        >
          <title>
            icon-
            copy
          </title>
        </svg>
      </button>
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
